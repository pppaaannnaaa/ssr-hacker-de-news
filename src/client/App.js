import React from 'react';
import PropTypes from 'prop-types';
import { renderRoutes } from 'react-router-config';
import Header from './components/Header';
import Footer from './components/Footer';
import ErrorBoundary from './components/ErrorBoundry';

const App = ({ route }) => {
  return (
    <div>
      <Header />
      <div className="container">
        <ErrorBoundary>{renderRoutes(route.routes)}</ErrorBoundary>
      </div>
      <Footer />
    </div>
  );
};

App.propTypes = {
  route: PropTypes.objectOf(PropTypes.any),
};

App.defaultProps = {
  route: null,
};

export default {
  component: App,
};
