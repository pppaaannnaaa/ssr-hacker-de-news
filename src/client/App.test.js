import React from 'react';
import { render, cleanup } from '@testing-library/react';
import '@testing-library/jest-dom/extend-expect';
import App from './App';

describe('<App />', () => {
  let props;

  beforeEach(() => {
    props = {
      route: {
        routes: [
          {
            ...App,
            path: '/',
            exact: true,
          },
        ],
      },
    };
  });

  afterEach(cleanup);

  test('Renders <App /> component correctly', () => {
    const { container, getByText } = render(<App.component {...props} />);
    expect(container).toMatchSnapshot();
    expect(getByText('News de Hacker')).toBeInTheDocument();
  });
});
