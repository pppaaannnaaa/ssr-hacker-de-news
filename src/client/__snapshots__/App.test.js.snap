// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<App /> Renders <App /> component correctly 1`] = `
<div>
  <div>
    <div
      aria-label="News de Hacker Logo"
      class="navbar-fixed"
      role="navigation"
    >
      <nav
        class="blue"
      >
        <div
          class="container"
        >
          <div
            class="nav-wrapper"
          >
            <a
              class="brand-logo"
              href="/"
              style="white-space: nowrap;"
            >
              News de Hacker
            </a>
          </div>
        </div>
      </nav>
    </div>
    <div
      class="container"
    >
      <div
        aria-label="Something went wrong"
        role="banner"
      >
        <h2>
          Something went wrong.
        </h2>
        <details
          style="white-space: pre-wrap;"
        >
          Error: Invariant failed: You should not use &lt;Switch&gt; outside a &lt;Router&gt;
          <br />
          
    in Switch (created by App)
    in ErrorBoundary (at App.js:13)
    in div (at App.js:12)
    in div (at App.js:10)
    in App (at App.test.js:26)
        </details>
      </div>
    </div>
    <footer
      aria-label="News de Hacker copyright"
      class="page-footer blue"
      role="navigation"
    >
      <div
        class="footer-copyright"
      >
        <div
          class="container"
        >
          Â© 2020 SSR News de Hacker
        </div>
      </div>
    </footer>
  </div>
</div>
`;
